src/scripts/main.js

```diff
@@ -1,3 +1,282 @@
+  board = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const tile = document.createElement('div');
+      const num = board[r][c];
+
+      tile.id = r.toString() + '-' + c.toString();
+      updateTile(tile, num);
+      gameField.append(tile);
+    }
+  }
+}
+
+button.addEventListener('click', e => {
+  if (button.classList.contains('start')) {
+    button.classList.replace('start', 'restart');
+    button.innerText = 'Restart';
+    startMessage.classList.add('hidden');
+
+    setTwoOrFour();
+    setTwoOrFour();
+  } else {
+    button.classList.replace('restart', 'start');
+    button.innerText = 'Start';
+    startMessage.classList.remove('hidden');
+    loseMessage.classList.add('hidden');
+
+    board = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ];
+
+    for (let r = 0; r < rows; r++) {
+      for (let c = 0; c < columns; c++) {
+        const tile = document.getElementById(`${r}-${c}`);
+
+        tile.innerText = '';
+        tile.classList.value = '';
+        tile.classList.add('field-cell');
+
+        winMessage.classList.add('hidden');
+      }
+    }
+
+    score = 0;
+    scoreGame.innerText = score;
+  }
+});
+
+function hasEmptyTile() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      if (board[r][c] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function get2048() {
+  for (let r = 0; r < rows; r++) {
+    for (let c = 0; c < columns; c++) {
+      const tile = document.getElementById(`${r}-${c}`);
+
+      if (tile.innerText === '2048') {
+        getNumber2048 = true;
+        winMessage.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+function setTwoOrFour() {
+  if (!hasEmptyTile()) {
+    return;
+  }
+
+  let found = false;
+
+  while (!found) {
+    const num = Math.random() < 0.1 ? 4 : 2;
+    const r = Math.floor(Math.random() * rows);
+    const c = Math.floor(Math.random() * columns);
+
+    if (board[r][c] === 0) {
+      board[r][c] = num;
+      found = true;
+
+      const tile = document.getElementById(r.toString() + '-' + c.toString());
+
+      tile.innerText = num; ;
+      tile.classList.add('field-cell--' + num.toString());
+    }
+  }
+}
+
+function updateTile(tile, num) {
+  tile.innerText = '';
+  tile.classList.value = '';
+  tile.classList.add('field-cell');
+
+  if (num > 0) {
+    tile.innerText = num;
+
+    if (num <= 2048) {
+      tile.classList.add('field-cell--' + num.toString());
+    }
+  }
+}
+
+document.addEventListener('keyup', (e) => {
+  if (getNumber2048) {
+    return;
+  }
+
+  if (e.code === 'ArrowLeft') {
+    slideLeft();
+    setTwoOrFour();
+  } else if (e.code === 'ArrowRight') {
+    slideRight();
+    setTwoOrFour();
+  } else if (e.code === 'ArrowUp') {
+    slideUp();
+    setTwoOrFour();
+  } else if (e.code === 'ArrowDown') {
```

Comment:
It is better to use switch...case here instead as you have a limited amount of codes